concordium-node/src/bin/cli.rs

```diff
@@ -62,25 +60,25 @@ async fn main() -> anyhow::Result<()> {
         instantiate_node(&conf, &mut app_prefs, stats_export_service, regenesis_arc.clone())
             .context("Failed to create the node.")?;
 
-    #[cfg(feature = "instrumentation")]
+    let (gen_data, priv_data) = get_baker_data(&app_prefs, &conf.cli.baker)
+        .context("Can't get genesis data or private data. Aborting")?;
+
+    // Start the prometheus server if the user requested it.
     {
-        let stats = node.stats.clone();
-        let pla = conf
-            .prometheus
-            .prometheus_listen_addr
-            .parse::<IpAddr>()
-            .context("Invalid Prometheus address")?;
-        let plp = conf.prometheus.prometheus_listen_port;
-        tokio::spawn(async move { stats.start_server(SocketAddr::new(pla, plp)).await });
+        if let Some(plp) = conf.prometheus.prometheus_listen_port {
+            let stats = node.stats.clone();
+            let pla = conf
+                .prometheus
+                .prometheus_listen_addr
+                .parse::<IpAddr>()
+                .context("Invalid Prometheus address")?;
+            tokio::spawn(async move { stats.start_server(SocketAddr::new(pla, plp)).await });
+        }
```

Comment:
Should we write a log stmt. that the prometheus server started?

Reply:
We can't really print it started. We can print it is starting.

Reply:
I added a log statement.